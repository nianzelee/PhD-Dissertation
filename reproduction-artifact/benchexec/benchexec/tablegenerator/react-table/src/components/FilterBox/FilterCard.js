// This file is part of BenchExec, a framework for reliable benchmarking:
// https://github.com/sosy-lab/benchexec
//
// SPDX-FileCopyrightText: 2019-2020 Dirk Beyer <https://www.sosy-lab.org>
//
// SPDX-License-Identifier: Apache-2.0

import React from "react";
import { faTrash } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Slider, { createSliderWithTooltip } from "rc-slider";
import "rc-slider/assets/index.css";

import {
  without,
  pathOr,
  emptyStateValue,
  getStep,
  NumberFormatterBuilder,
} from "../../utils/utils";
import { statusForEmptyRows } from "../../utils/filters";

const Range = createSliderWithTooltip(Slider.Range);

const numericInputDebounce = 500;
let debounceHandler = setTimeout(() => {}, numericInputDebounce);

export default class FilterCard extends React.PureComponent {
  constructor(props) {
    super(props);
    const {
      values,
      min,
      max,
      type,
      number_of_significant_digits: significantDigits,
    } = props.filter || { values: [] };
    let sliderMin = 0;
    let sliderMax = 0;
    if (type === "measure" || type === "number") {
      const builder = new NumberFormatterBuilder(significantDigits).build();
      sliderMin = builder(min);
      sliderMax = builder(max);
      const value = values && values[0];
      if (value && value.includes(":")) {
        const res = this.handleMinMaxValue(value, significantDigits);
        sliderMin = res.min;
        sliderMax = res.max;
      }
    }
    this.state = {
      title:
        props.availableFilters && props.availableFilters.length
          ? props.availableFilters[0].title
          : "",
      values: [],
      idx: pathOr(["availableFilters", 0, "idx"], 0, props),
      active: true,
      selectedDistincts: [],
      sliderMin,
      sliderMax,
      numericMin: null,
      numericMax: null,
    };
  }

  sendFilterUpdate(values) {
    const { type, categories } = this.props.filter;
    if (
      categories &&
      categories.includes("empty ") &&
      !values.includes(statusForEmptyRows)
    ) {
      values = values.concat(statusForEmptyRows);
    }
    if (values.length === 0 && type === "status") {
      this.props.onFilterUpdate({
        values: [emptyStateValue],
        title: this.state.title || this.props.title,
      });
    } else {
      this.props.onFilterUpdate({
        values,
        title: this.state.title || this.props.title,
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (!this.props.filter) {
      return;
    }
    if (
      !prevProps.filter ||
      prevProps.filter.values !== this.props.filter.values
    ) {
      const {
        values,
        number_of_significant_digits: significantDigits,
      } = this.props.filter;
      const [value] = values;
      if (value && value.includes(":")) {
        const { min, max } = this.handleMinMaxValue(value, significantDigits);
        this.setState({
          sliderMin: min,
          sliderMax: max,
          numericMin: min,
          numericMax: max,
        });
      }
    }
  }

  handleMinMaxValue(value, significantDigits) {
    const builder = new NumberFormatterBuilder(significantDigits).build();
    const { min: propMin, max: propMax } = this.props.filter || {
      min: 0,
      max: Infinity,
    };
    const [vMin, vMax] = value.split(":");
    return {
      min: vMin.trim() !== "" ? vMin : builder(propMin),
      max: vMax.trim() !== "" ? vMax : builder(propMax),
    };
  }

  handleNumberChange(min, max) {
    const newState = {};
    newState.sliderMin = Number(this.state.numericMin ?? this.state.sliderMin);
    newState.sliderMax = Number(this.state.numericMax ?? this.state.sliderMax);
    if (newState.sliderMin > newState.sliderMax) {
      const temp = newState.sliderMax;
      newState.sliderMax = newState.sliderMin;
      newState.sliderMin = temp;
    }
    // defaulting to an empty string per side, if the values exceeds
    // or is less than the min/max thresholds
    const stringRepMin =
      newState.sliderMin <= Number(min) ? "" : newState.sliderMin;
    const stringRepMax =
      newState.sliderMax >= Number(max) ? "" : newState.sliderMax;
    newState.values = [`${stringRepMin}:${stringRepMax}`];
    this.setState(newState);
    this.sendFilterUpdate(newState.values);
  }

  render() {
    const { filter, editable, availableFilters } = this.props;
    const selectRef = React.createRef();
    const filterAddSelection = () => (
      <>
        <span style={{ marginLeft: "12px" }}>Add filter for: </span>
        <select
          className="filter-selection"
          defaultValue="-1"
          ref={selectRef}
          onChange={({ target: { value: idx } }) => {
            if (idx === -1) {
              return;
            }
            this.setState({ idx: -1, active: true });
            selectRef.current.value = "-1"; // Reset preselected option to "Column"
            this.props.addFilter(idx);
          }}
        >
          <option value="-1" disabled>
            Column
          </option>
          {availableFilters.map(({ idx, display_title }) => (
            <option key={idx} value={idx}>
              {display_title}
            </option>
          ))}
        </select>
      </>
    );

    const makeHeader = (name, editable) => (
      <div className="filter-card--header">
        {editable ? (
          filterAddSelection()
        ) : (
          <>
            <h4 className="title">{`${filter.display_title} ${
              filter.unit ? "(" + filter.unit + ")" : ""
            }`}</h4>
            <FontAwesomeIcon
              className="delete-button"
              icon={faTrash}
              onClick={() => {
                this.props.removeFilter();
              }}
            />
          </>
        )}
      </div>
    );

    const makeFilterBody = (filter) => {
      if (!filter) {
        return null;
      }
      const {
        title,
        type,
        number_of_significant_digits: significantDigits,
        categories,
        statuses,
        values = [],
      } = filter;
      let { min, max } = filter;
      let body;
      const emptyRowRef = React.createRef();
      if (type === "status") {
        body = (
          <>
            {this.props.filter.categories &&
              this.props.filter.categories.includes("empty ") && (
                <div className="filter-card--body--empty-rows">
                  Empty rows{" "}
                  <input
                    type="checkbox"
                    name={`empty-rows`}
                    ref={emptyRowRef}
                    checked={values.includes("empty ")}
                    onChange={({ target: { checked } }) => {
                      const emptyValue = "empty ";
                      if (checked) {
                        const newValues = [...values, emptyValue];
                        this.setState({ values: newValues });
                        this.sendFilterUpdate(newValues);
                      } else {
                        const newValues = without(emptyValue, values);

                        this.setState({ values: newValues });
                        this.sendFilterUpdate(newValues);
                      }
                    }}
                  />
                </div>
              )}
            Category
            <ul className="filter-card--body--list">
              {categories
                .filter((category) => category !== "empty ")
                .sort()
                .map((category) => {
                  const ref = React.createRef();
                  return (
                    <li key={category}>
                      <input
                        type="checkbox"
                        name={`cat-${category}`}
                        checked={values.includes(category)}
                        ref={ref}
                        onChange={({ target: { checked } }) => {
                          if (checked) {
                            const newValues = [...values, category];
                            this.setState({ values: newValues });
                            this.sendFilterUpdate(newValues);
                          } else {
                            const newValues = without(category, values);

                            this.setState({ values: newValues });
                            this.sendFilterUpdate(newValues);
                          }
                        }}
                      />
                      <label
                        htmlFor={`cat-${category}`}
                        onClick={() => ref.current.click()}
                      >
                        {category}
                      </label>
                    </li>
                  );
                })}
            </ul>
            Status
            <ul className="filter-card--body--list">
              {statuses.sort().map((status) => {
                const ref = React.createRef();
                return (
                  <li key={status}>
                    <input
                      type="checkbox"
                      name={`stat-${status}`}
                      ref={ref}
                      checked={values.includes(status)}
                      onChange={({ target: { checked } }) => {
                        if (checked) {
                          const newValues = [...values, status];
                          this.setState({ values: newValues });
                          this.sendFilterUpdate(newValues);
                        } else {
                          const newValues = without(status, values);
                          this.setState({ values: newValues });
                          this.sendFilterUpdate(newValues);
                        }
                      }}
                    />
                    <label
                      htmlFor={`stat-${status}`}
                      onClick={() => ref.current.click()}
                    >
                      {status}
                    </label>
                  </li>
                );
              })}
            </ul>
          </>
        );
      } else if (type === "text") {
        const [value] = values;

        body = (
          <input
            type="text"
            name={`text-${title}`}
            placeholder="Search for value"
            value={value}
            onChange={({ target: { value: textValue } }) => {
              clearTimeout(debounceHandler);
              this.setState({ values: [textValue] });
              debounceHandler = setTimeout(() => {
                this.sendFilterUpdate([textValue]);
              }, numericInputDebounce);
            }}
          />
        );
      } else {
        const builder = new NumberFormatterBuilder(significantDigits).build();
        min = builder(min);
        max = builder(max);
        const minStep = getStep(min);
        const maxStep = getStep(max);

        // get the bigger step by length of string (== smaller step)
        const step = minStep.length > maxStep.length ? minStep : maxStep;

        //shift the decimal
        body = (
          <>
            <div className="filter-card--range-container">
              <b>{min}</b>
              <b>{max}</b>
            </div>
            <Range
              min={Number(min)}
              max={Number(max)}
              step={step}
              defaultValue={[Number(min), Number(max)]}
              value={[
                Number(this.state.sliderMin),
                Number(this.state.sliderMax),
              ]}
              onChange={([nMin, nMax]) => {
                this.setState({
                  sliderMin: builder(nMin),
                  sliderMax: builder(nMax),
                });
              }}
              onAfterChange={([nMin, nMax]) => {
                const fMin = builder(nMin);
                const fMax = builder(nMax);
                const stringRepMin = fMin === min ? "" : fMin;
                const stringRepMax = fMax === max ? "" : fMax;
                this.setState({
                  sliderMin: fMin,
                  sliderMax: fMax,
                  numericMin: nMin,
                  numericMax: nMax,
                  values: [`${stringRepMin}:${stringRepMax}`],
                });
                this.sendFilterUpdate([`${stringRepMin}:${stringRepMax}`]);
              }}
            />
            <div className="filter-card--range-input-fields">
              <label
                className="range-input-fields--min"
                htmlFor={`inp-${title}-min`}
              >
                minimum
              </label>
              <label
                className="range-input-fields--max"
                htmlFor={`inp-${title}-max`}
              >
                maximum
              </label>
              <input
                type="number"
                name={`inp-${title}-min`}
                value={
                  this.state.numericMin !== null
                    ? this.state.numericMin
                    : this.state.sliderMin
                }
                lang="en-US"
                step={step}
                onChange={({ target: { value } }) => {
                  if (this.numericMinTimeout) {
                    clearTimeout(this.numericMinTimeout);
                  }
                  this.setState({ numericMin: value });
                  this.numericMinTimeout = setTimeout(
                    () => this.handleNumberChange(min, max),
                    numericInputDebounce,
                  );
                }}
              />
              <input
                type="number"
                name={`inp-${title}-max`}
                step={step}
                lang="en-US"
                value={
                  this.state.numericMax !== null
                    ? this.state.numericMax
                    : this.state.sliderMax
                }
                onChange={({ target: { value } }) => {
                  if (this.numericMaxTimeout) {
                    clearTimeout(this.numericMaxTimeout);
                  }
                  this.setState({ numericMax: value });
                  this.numericMaxTimeout = setTimeout(
                    () => this.handleNumberChange(min, max),
                    numericInputDebounce,
                  );
                }}
              />
            </div>
          </>
        );
      }
      return <div className="filter-card--body">{body}</div>;
    };

    return (
      <div className="filter-card">
        {makeHeader(this.props.name, editable)}
        {makeFilterBody(this.props.filter)}
      </div>
    );
  }
}
