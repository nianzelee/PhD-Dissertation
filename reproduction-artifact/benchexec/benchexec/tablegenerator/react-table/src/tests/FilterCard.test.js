// This file is part of BenchExec, a framework for reliable benchmarking:
// https://github.com/sosy-lab/benchexec
//
// SPDX-FileCopyrightText: 2019-2020 Dirk Beyer <https://www.sosy-lab.org>
//
// SPDX-License-Identifier: Apache-2.0

import React from "react";
import renderer from "react-test-renderer";
import FilterCard from "../components/FilterBox/FilterCard.js";

import { shallow } from "enzyme";

const createFilterCard = (props) => <FilterCard {...props} />;

describe("FilterCard tests", () => {
  test("FilterCard should allow selection of available filters", () => {
    const availableFilters = [
      { display_title: "Filter1", idx: 0 },
      { display_title: "Filter2", idx: 1 },
    ];
    const Card = createFilterCard({ availableFilters, editable: true });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should display checkboxes for status filters", () => {
    const filter = {
      display_title: "Status",
      categories: ["cat1", "correct", "wrong"],
      statuses: ["true", "false(reach)"],
      type: "status",
    };
    const Card = createFilterCard({ title: "Status", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should display range slider for number filters", () => {
    const filter = {
      display_title: "cputime",
      min: 1337,
      max: 9001,
      type: "measure",
    };
    const Card = createFilterCard({ title: "cputime", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should display textbox for text filters", () => {
    const filter = {
      display_title: "host",
      type: "text",
    };
    const Card = createFilterCard({ title: "host", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should correctly reflect already set status filters", () => {
    const filter = {
      display_title: "Status",
      categories: ["cat1", "correct", "wrong"],
      statuses: ["true", "false(reach)"],
      values: ["true"],
      type: "status",
    };
    const Card = createFilterCard({ title: "Status", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should correctly reflect already set number filters", () => {
    const filter = {
      display_title: "cputime",
      min: 1337,
      max: 9001,
      values: ["1500:3000"],
      type: "measure",
    };
    const Card = createFilterCard({ title: "cputime", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should correctly reflect already set text filters", () => {
    const filter = {
      display_title: "host",
      type: "text",
      values: ["node-"],
    };
    const Card = createFilterCard({ title: "host", filter });

    const component = renderer.create(Card);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("FilterCard should send correct updates on selection of filters", () => {
    let response = {};

    const handler = (obj) => {
      response = obj;
    };
    const filter = {
      display_title: "Status",
      categories: ["cat1", "correct", "wrong"],
      statuses: ["true", "false(reach)"],
      type: "status",
    };
    const Card = createFilterCard({
      title: "Status",
      filter,
      onFilterUpdate: handler,
    });
    const wrapper = shallow(Card);

    const checkBox = wrapper.findWhere(
      (node) => node.type() === "input" && node.props().name === "stat-true",
    );

    expect(checkBox).toHaveLength(1);

    checkBox.simulate("change", { target: { checked: true } });

    expect(response).toEqual({ values: ["true"], title: "Status" });
  });
});
