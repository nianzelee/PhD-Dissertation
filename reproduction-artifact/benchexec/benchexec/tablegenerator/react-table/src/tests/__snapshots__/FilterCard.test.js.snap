// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterCard tests FilterCard should allow selection of available filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <span
      style={
        Object {
          "marginLeft": "12px",
        }
      }
    >
      Add filter for: 
    </span>
    <select
      className="filter-selection"
      defaultValue="-1"
      onChange={[Function]}
    >
      <option
        disabled={true}
        value="-1"
      >
        Column
      </option>
      <option
        value={0}
      >
        Filter1
      </option>
      <option
        value={1}
      >
        Filter2
      </option>
    </select>
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should correctly reflect already set number filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      cputime 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    <div
      className="filter-card--range-container"
    >
      <b>
        1337
      </b>
      <b>
        9001
      </b>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
      />
      <div
        className="rc-slider-track rc-slider-track-1"
        style={
          Object {
            "left": "2.1268267223382047%",
            "right": "auto",
            "width": "19.572025052192064%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={9001}
        aria-valuemin={1337}
        aria-valuenow={1500}
        className="rc-slider-handle rc-slider-handle-1"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "2.1268267223382047%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        aria-disabled={false}
        aria-valuemax={9001}
        aria-valuemin={1337}
        aria-valuenow={3000}
        className="rc-slider-handle rc-slider-handle-2"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "21.69885177453027%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="filter-card--range-input-fields"
    >
      <label
        className="range-input-fields--min"
        htmlFor="inp-undefined-min"
      >
        minimum
      </label>
      <label
        className="range-input-fields--max"
        htmlFor="inp-undefined-max"
      >
        maximum
      </label>
      <input
        lang="en-US"
        name="inp-undefined-min"
        onChange={[Function]}
        step={1}
        type="number"
        value="1500"
      />
      <input
        lang="en-US"
        name="inp-undefined-max"
        onChange={[Function]}
        step={1}
        type="number"
        value="3000"
      />
    </div>
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should correctly reflect already set status filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      Status 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    Category
    <ul
      className="filter-card--body--list"
    >
      <li>
        <input
          checked={false}
          name="cat-cat1"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-cat1"
          onClick={[Function]}
        >
          cat1
        </label>
      </li>
      <li>
        <input
          checked={false}
          name="cat-correct"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-correct"
          onClick={[Function]}
        >
          correct
        </label>
      </li>
      <li>
        <input
          checked={false}
          name="cat-wrong"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-wrong"
          onClick={[Function]}
        >
          wrong
        </label>
      </li>
    </ul>
    Status
    <ul
      className="filter-card--body--list"
    >
      <li>
        <input
          checked={false}
          name="stat-false(reach)"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="stat-false(reach)"
          onClick={[Function]}
        >
          false(reach)
        </label>
      </li>
      <li>
        <input
          checked={true}
          name="stat-true"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="stat-true"
          onClick={[Function]}
        >
          true
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should correctly reflect already set text filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      host 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    <input
      name="text-undefined"
      onChange={[Function]}
      placeholder="Search for value"
      type="text"
      value="node-"
    />
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should display checkboxes for status filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      Status 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    Category
    <ul
      className="filter-card--body--list"
    >
      <li>
        <input
          checked={false}
          name="cat-cat1"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-cat1"
          onClick={[Function]}
        >
          cat1
        </label>
      </li>
      <li>
        <input
          checked={false}
          name="cat-correct"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-correct"
          onClick={[Function]}
        >
          correct
        </label>
      </li>
      <li>
        <input
          checked={false}
          name="cat-wrong"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="cat-wrong"
          onClick={[Function]}
        >
          wrong
        </label>
      </li>
    </ul>
    Status
    <ul
      className="filter-card--body--list"
    >
      <li>
        <input
          checked={false}
          name="stat-false(reach)"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="stat-false(reach)"
          onClick={[Function]}
        >
          false(reach)
        </label>
      </li>
      <li>
        <input
          checked={false}
          name="stat-true"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="stat-true"
          onClick={[Function]}
        >
          true
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should display range slider for number filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      cputime 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    <div
      className="filter-card--range-container"
    >
      <b>
        1337
      </b>
      <b>
        9001
      </b>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
      />
      <div
        className="rc-slider-track rc-slider-track-1"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "100%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={9001}
        aria-valuemin={1337}
        aria-valuenow={1337}
        className="rc-slider-handle rc-slider-handle-1"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        aria-disabled={false}
        aria-valuemax={9001}
        aria-valuemin={1337}
        aria-valuenow={9001}
        className="rc-slider-handle rc-slider-handle-2"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "100%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="filter-card--range-input-fields"
    >
      <label
        className="range-input-fields--min"
        htmlFor="inp-undefined-min"
      >
        minimum
      </label>
      <label
        className="range-input-fields--max"
        htmlFor="inp-undefined-max"
      >
        maximum
      </label>
      <input
        lang="en-US"
        name="inp-undefined-min"
        onChange={[Function]}
        step={1}
        type="number"
        value="1337"
      />
      <input
        lang="en-US"
        name="inp-undefined-max"
        onChange={[Function]}
        step={1}
        type="number"
        value="9001"
      />
    </div>
  </div>
</div>
`;

exports[`FilterCard tests FilterCard should display textbox for text filters 1`] = `
<div
  className="filter-card"
>
  <div
    className="filter-card--header"
  >
    <h4
      className="title"
    >
      host 
    </h4>
    <i
      className="fa trash"
    />
  </div>
  <div
    className="filter-card--body"
  >
    <input
      name="text-undefined"
      onChange={[Function]}
      placeholder="Search for value"
      type="text"
    />
  </div>
</div>
`;
